modules/unsupported/geojson-core/src/main/java/org/geotools/filter/expression/JSONNodePropertyAccessorFactory.java

```diff
@@ -0,0 +1,212 @@
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.filter.expression;
+
+import com.fasterxml.jackson.core.JsonPointer;
+import com.fasterxml.jackson.databind.JsonNode;
+import java.time.Instant;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.TemporalAccessor;
+import java.util.Date;
+import java.util.Map;
+import org.geotools.data.geojson.GeoJSONReader;
+import org.geotools.util.factory.Hints;
+import org.opengis.feature.Property;
+import org.opengis.feature.simple.SimpleFeature;
+
+/** Property Accessor Factory for {@link JsonNode} */
+public class JSONNodePropertyAccessorFactory implements PropertyAccessorFactory {
+
+    static PropertyAccessor JSONNODEPROPERTY = new JSONNodePropertyAccessor();
+    static final String ROOT = "/";
+
+    @Override
+    public PropertyAccessor createPropertyAccessor(
+            Class<?> type, String xpath, Class<?> target, Hints hints) {
+        if (xpath == null) {
+            return null;
+        }
+        return JSONNODEPROPERTY;
+    }
+
+    /** Property Accessor for {@link JsonNode} */
+    static class JSONNodePropertyAccessor implements PropertyAccessor {
+
+        private static final String JSON_NODE_DATE_FORMAT_OFF = "JSON_NODE_DATE_FORMAT_OFF";
+
+        @Override
+        public boolean canHandle(Object object, String xpath, Class target) {
+            try {
+                if (xpath != null && object != null && object instanceof SimpleFeature) {
+                    String[] parts = stripAndReturnHeadAndRest(xpath);
+                    SimpleFeature feature = (SimpleFeature) object;
+                    Property property = feature.getProperty(parts[0]);
+                    Object value = null;
+                    if (property == null) {
+                        // check to see if it is in the user data map which is sometimes populated
+                        // from geojson
+                        value = getValueFromUserData(parts, feature, value);
+                    } else {
+                        value = property.getValue();
+                    }
+                    if (value != null && value instanceof JsonNode) {
+                        if (parts.length < 2) {
+                            // only one part means return whole JsonNode
+                            return true;
+                        } else {
+                            JsonPointer pointer = JsonPointer.compile(parts[1]);
+                            JsonNode jsonNode = ((JsonNode) property.getValue()).at(pointer);
+                            return !jsonNode.isMissingNode();
+                        }
+                    }
+                }
```

Comment:
Most of this code is duplicated below. Can you extract a shared method that would extract the target jsonNode or null, and then use it in a different way in canHandle and in get?

Reply:
Extracted into seperate method